sklearn/decomposition/tests/test_pca.py

```diff
@@ -529,6 +529,26 @@ def test_pca_score3():
     assert_true(ll.argmax() == 1)
 
 
+def test_pca_score4():
```

Comment:
You can keep this test but I feel there must be a simpler test. I am going to think about it a bit more and come back to you.

A simpler test more related to your fix would be to check that `noise_variance_` is 0 when either `n_components < n_features` or `n_components < n_samples`.